### PR TITLE
Targets not required in promtail config

### LINKED ISSUE #1929: `targets` should not be required for promtail scrape config
```
scrape_configs:
- job_name: system
  static_configs:
  - targets:
      - localhost
    labels:
      job: varlogs
      __path__: /var/log/*log

```

The `targets` definition is required because of the use of Prometheus code however has no real use or application in Promtail.

Not sure how easy or possible it is to remove this but it would be a nice simplification on the config.

Relevant code here:

https://github.com/grafana/loki/blob/5e752bf343e221b274319074d358cd4dceca0bb8/pkg/promtail/targets/filetargetmanager.go#L208

### PR DESCRIPTION
**Pull request by adityacs**
**What this PR does / why we need it**:
Targets are not required in promtail config. If defined in config we will just use that otherwise we will just add `dummy` host.

**Which issue(s) this PR fixes**:
Fixes #1929 

**Checklist**
- [x] Documentation added
- [ ] Tests updated

### CODE REVIEW

**Comment by adityacs** [comment]
@slim-bean Does this change has any impact on docker driver and any other module?  I am not aware of other uses of this. Hope this change doesn't break anything

File: pkg/promtail/targets/filetargetmanager.go [hunk 1]
```diff
@@ -109,9 +109,15 @@ func NewFileTargetManager(
 		}
 
 		// Add Source value to the static config target groups for unique identification
-		// within scrape pool.
+		// within scrape pool. Also, set dummy target if target is not defined in promtail config.
+		// Just to make sure prometheus target group sync works fine.
 		for i, tg := range cfg.ServiceDiscoveryConfig.StaticConfigs {
 			tg.Source = fmt.Sprintf("%d", i)
+			if !(len(tg.Targets) > 0) {
```
Comment (owen-d) [inline pkg/promtail/targets/filetargetmanager.go:116]:
```suggestion
			if len(tg.Targets) == 0 {
```

Comment (adityacs) [reply pkg/promtail/targets/filetargetmanager.go:116]: Perfect. Thanks @owen-d

**Comment by adityacs** [comment]
@owen-d We don't have to name it `dummy`, we can make it `localhost`. But we should assign some target name otherwise `target` sync code will be skipped and file targets won't be initialized. 

Also, I am completely for "removing this(targets) silently from the docs". Just that it should not confuse the existing users. 

> However this would create difficulties trying to copy/paste prometheus service discovery code into promtail.

Agree on this but I think users will eventually get adjusted to this

**Comment by slim-bean** [comment]
Wanted to get this into the 1.5 RC so I inserted my opinions via `git push` :)

I think using `localhost` instead of `dummy` will be more intuitive for people and less confusing/alarming if they are looking at the promtail /targets and /service-discovery http pages.

I also re-worded the docs a bit.

**Comment by adityacs** [comment]
Thank you @slim-bean

**Pull request merged**